client/controller/shared.js

```diff
@@ -79,3 +83,61 @@ export function composeHandlers( ...handlers ) {
 		handleNext();
 	};
 }
+
+const getLocalizedCanonicalUrl = ( path, locale ) => {
+	const baseUrl = `https://wordpress.com${ path }`;
+	const baseUrlWithoutLang = baseUrl.replace(
+		new RegExp( `\\/(${ getLanguageSlugs().join( '|' ) })(\\/|\\?|$)` ),
+		'$2'
+	);
+	let localizedUrl = localizeUrl( baseUrlWithoutLang, locale, false );
+
+	// Remove the trailing slash if `path` doesn't have one either.
+	if ( path.slice( -1 ) !== '/' && localizedUrl.slice( -1 ) === '/' ) {
```

Comment:
You can write `path.endsWith( '/' )`, that better conveys the intent.

Reply:
Fixed, thanks!